From c7f4f02ea7ff061fecc5ddd3c948ac6e7df6cae2 Mon Sep 17 00:00:00 2001
From: Kouei Abe <kouei.abe.cp@rms.renesas.com>
Date: Mon, 12 Nov 2012 09:16:37 +0900
Subject: [PATCH 082/215] dma: hbp-dmae: Remove verbose function

Signed-off-by: Kouei Abe <kouei.abe.cp@rms.renesas.com>
Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 drivers/dma/rcar-hpbdma.c |   14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/drivers/dma/rcar-hpbdma.c b/drivers/dma/rcar-hpbdma.c
index 7c60e78..30a673f 100644
--- a/drivers/dma/rcar-hpbdma.c
+++ b/drivers/dma/rcar-hpbdma.c
@@ -99,16 +99,6 @@ static void dmadintcr_write(struct hpb_dmae_device *hpbdev, u32 ch)
 			 / sizeof(u32));
 }
 
-static void dmaasyncmdr_write(struct hpb_dmae_device *hpbdev, u32 data)
-{
-	__raw_writel(data, hpbdev->mode_reg);
-}
-
-static u32 dmaasyncmdr_read(struct hpb_dmae_device *hpbdev)
-{
-	return __raw_readl(hpbdev->mode_reg);
-}
-
 static void dmae_enable_int(struct hpb_dmae_device *hpbdev, u32 dmach)
 {
 	int intreg;
@@ -154,10 +144,10 @@ static void dmae_set_async_mode(struct hpb_dmae_device *hpbdev,
 	u32 mode;
 
 	spin_lock_bh(&hpbdev->reg_lock);
-	mode = dmaasyncmdr_read(hpbdev);
+	mode = __raw_readl(hpbdev->mode_reg);
 	mode &= ~mask;
 	mode |= data;
-	dmaasyncmdr_write(hpbdev, mode);
+	__raw_writel(mode, hpbdev->mode_reg);
 	spin_unlock_bh(&hpbdev->reg_lock);
 }
 
-- 
1.7.9.5

