From 5b62f2038c036e820615de68e9a1ae0ebcebcb86 Mon Sep 17 00:00:00 2001
From: Kouei Abe <kouei.abe.cp@rms.renesas.com>
Date: Mon, 26 Nov 2012 15:05:30 +0900
Subject: [PATCH 083/215] dma: hpb-dmae: Remove verbose return value

Signed-off-by: Kouei Abe <kouei.abe.cp@rms.renesas.com>
Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 drivers/dma/rcar-hpbdma.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/dma/rcar-hpbdma.c b/drivers/dma/rcar-hpbdma.c
index 30a673f..7d6d96a 100644
--- a/drivers/dma/rcar-hpbdma.c
+++ b/drivers/dma/rcar-hpbdma.c
@@ -235,11 +235,9 @@ static void dmae_set_port(struct hpb_dmae_chan *hpb_chan, u32 portreg)
 	hpb_chan->xmit_shift = calc_xmit_shift(hpb_chan);
 }
 
-static int dmae_set_dcr(struct hpb_dmae_chan *hpb_chan, u32 val)
+static void dmae_set_dcr(struct hpb_dmae_chan *hpb_chan, u32 val)
 {
 	hpb_dmae_writel(hpb_chan, val, DCR);
-
-	return 0;
 }
 
 static dma_cookie_t hpb_dmae_tx_submit(struct dma_async_tx_descriptor *tx)
-- 
1.7.9.5

