From a5beb753b73f682b24bd5e71cec83333e5cdde8b Mon Sep 17 00:00:00 2001
From: Kouei Abe <kouei.abe.cp@rms.renesas.com>
Date: Mon, 12 Nov 2012 09:17:28 +0900
Subject: [PATCH 084/215] dma: hpb-dmae: Remove useless code

Signed-off-by: Kouei Abe <kouei.abe.cp@rms.renesas.com>
Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 drivers/dma/rcar-hpbdma.c |   10 ----------
 1 file changed, 10 deletions(-)

diff --git a/drivers/dma/rcar-hpbdma.c b/drivers/dma/rcar-hpbdma.c
index 7d6d96a..bd7cf48 100644
--- a/drivers/dma/rcar-hpbdma.c
+++ b/drivers/dma/rcar-hpbdma.c
@@ -165,17 +165,7 @@ static int hpb_dmae_rst(struct hpb_dmae_device *hpbdev)
 
 	return 0;
 }
-#if 0
-static bool dmae_is_busy(struct hpb_dmae_chan *hpb_chan)
-{
-	u32 dstsr = hpb_dmae_readl(hpb_chan, DSTSR);
-
-	if ((dstsr & 0x01) == 0x01)
-		return true; /* working */
 
-	return false; /* waiting */
-}
-#endif
 static unsigned int calc_xmit_shift(struct hpb_dmae_chan *hpb_chan)
 {
 	struct hpb_dmae_device *hpbdev = container_of(hpb_chan->common.device,
-- 
1.7.9.5

