From 8f3e842e94a3a3c7d7f866fb1e948f1ac5f108d0 Mon Sep 17 00:00:00 2001
From: Phil Edworthy <phil.edworthy@renesas.com>
Date: Wed, 21 Nov 2012 09:32:32 +0000
Subject: [PATCH 098/215] marzen: Add HSPI2B for SPI Flash

Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 arch/arm/mach-shmobile/board-marzen.c |   22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm/mach-shmobile/board-marzen.c b/arch/arm/mach-shmobile/board-marzen.c
index 041396d..babcf9c 100644
--- a/arch/arm/mach-shmobile/board-marzen.c
+++ b/arch/arm/mach-shmobile/board-marzen.c
@@ -155,6 +155,21 @@ static struct platform_device hspi_device = {
 	.num_resources	= ARRAY_SIZE(hspi_resources),
 };
 
+static struct resource hspi2_resources[] = {
+	[0] = {
+		.start		= 0xFFFC6000,
+		.end		= 0xFFFC6018 - 1,
+		.flags		= IORESOURCE_MEM,
+	},
+};
+
+static struct platform_device hspi2_device = {
+	.name	= "sh-hspi",
+	.id	= 2,
+	.resource	= hspi2_resources,
+	.num_resources	= ARRAY_SIZE(hspi2_resources),
+};
+
 static struct resource rcar_du0_resources[] = {
 	[0] = {
 		.name	= "Display Unit 0",
@@ -670,6 +685,7 @@ static struct platform_device *marzen_devices[] __initdata = {
 	&sdhi0_device,
 	&thermal_device,
 	&hspi_device,
+	&hspi2_device,
 	&ehci0_device,
 	&ohci0_device,
 	&ehci1_device,
@@ -742,6 +758,12 @@ static void __init marzen_init(void)
 	gpio_request(GPIO_FN_HSPI_TX0,	NULL);
 	gpio_request(GPIO_FN_HSPI_RX0,	NULL);
 
+	/* HSPI 2 B (SPI Flash) */
+	gpio_request(GPIO_FN_HSPI_CLK2_B, NULL);
+	gpio_request(GPIO_FN_HSPI_CS2_B, NULL);
+	gpio_request(GPIO_FN_HSPI_TX2_B, NULL);
+	gpio_request(GPIO_FN_HSPI_RX2_B, NULL);
+
 	/* USB (CN21) */
 	gpio_request(GPIO_FN_USB_OVC0, NULL);
 	gpio_request(GPIO_FN_USB_OVC1, NULL);
-- 
1.7.9.5

