From 9b822c60492eaa9e947c811b9d0d06cbd1db5a24 Mon Sep 17 00:00:00 2001
From: Phil Edworthy <phil.edworthy@renesas.com>
Date: Wed, 21 Nov 2012 15:54:03 +0000
Subject: [PATCH 099/215] marzen: Register spidev devices

Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 arch/arm/mach-shmobile/board-marzen.c |   18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm/mach-shmobile/board-marzen.c b/arch/arm/mach-shmobile/board-marzen.c
index babcf9c..07d88d0 100644
--- a/arch/arm/mach-shmobile/board-marzen.c
+++ b/arch/arm/mach-shmobile/board-marzen.c
@@ -170,6 +170,21 @@ static struct platform_device hspi2_device = {
 	.num_resources	= ARRAY_SIZE(hspi2_resources),
 };
 
+static struct spi_board_info marzen_spi_devices[] __initdata = {
+	{
+		.modalias = "spidev",
+		.max_speed_hz = 100000000,
+		.bus_num = 0,
+		.chip_select = 0,
+	},
+	{
+		.modalias = "spidev",
+		.max_speed_hz = 100000000,
+		.bus_num = 2,
+		.chip_select = 0,
+	},
+};
+
 static struct resource rcar_du0_resources[] = {
 	[0] = {
 		.name	= "Display Unit 0",
@@ -844,6 +859,9 @@ static void __init marzen_init(void)
 	i2c_register_board_info(0, marzen_i2c_devices,
 				ARRAY_SIZE(marzen_i2c_devices));
 
+	spi_register_board_info(marzen_spi_devices,
+		ARRAY_SIZE(marzen_spi_devices));
+
 	rcar_usbh_init();
 }
 
-- 
1.7.9.5

