From f6381c240456aa94bf14566026a257710b9799fb Mon Sep 17 00:00:00 2001
From: Phil Edworthy <phil.edworthy@renesas.com>
Date: Tue, 11 Dec 2012 16:08:45 +0000
Subject: [PATCH 106/215] sata: rcar: Remove duplicate entry in Kconfig

Signed-off-by: Phil Edworthy <phil.edworthy@renesas.com>
---
 drivers/ata/Kconfig |    8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/ata/Kconfig b/drivers/ata/Kconfig
index ad6a550..ebd7a06 100644
--- a/drivers/ata/Kconfig
+++ b/drivers/ata/Kconfig
@@ -299,14 +299,6 @@ config SATA_RCAR
 
 	  If unsure, say N.
 
-config SATA_RCAR
-	tristate "R-Car SATA support"
-	help
-	  This option enables support for the Renesas R-Car series
-	  Serial ATA controller.
-
-	  If unsure, say N.
-
 comment "PATA SFF controllers with BMDMA"
 
 config PATA_ALI
-- 
1.7.9.5

